### PR TITLE
APM > .NET > Add initial DBM configuration key to APM configuration page.

### PR DESCRIPTION
**Pull request by zacharycmontoya**
### What does this PR do?
1. Adds setup instructions to the DBM <-> APM link guide for compatibility with .NET
2. Adds the DBM configuration setting `DD_DBM_PROPAGATION_MODE` to the .NET tracing configuration pages

### Motivation
.NET has recently implemented the ability to link spans to DBM.

### Preview
DBM guide page: https://docs-staging.datadoghq.com/zach.montoya/apm-dotnet-configuration-dbm/database_monitoring/guide/connect_dbm_and_apm/?tab=net#setup

.NET Framework APM configuration page: https://docs-staging.datadoghq.com/zach.montoya/apm-dotnet-configuration-dbm/tracing/trace_collection/library_config/dotnet-framework/?tab=environmentvariables#automatic-instrumentation-optional-configuration

.NET Core APM configuration page: https://docs-staging.datadoghq.com/zach.montoya/apm-dotnet-configuration-dbm/tracing/trace_collection/library_config/dotnet-core/?tab=environmentvariables#automatic-instrumentation-optional-configuration

### Additional Notes
N/A

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/database_monitoring/guide/connect_dbm_and_apm.md [hunk 1]
```diff
@@ -264,6 +267,29 @@ client.query('SELECT $1::text as message', ['Hello world!'], (err, result) => {
...
+
+Ensure that you are using a supported client library (e.g. `Npgsql`).
+
+Enable the database monitoring propagation feature by setting the following environment variable:
+   - `DD_DBM_PROPAGATION_MODE=full`
+
+For the best user experience ensure the following environment variables are set in your application:
+   - `DD_SERVICE=(application name)`
+   - `DD_ENV=(application environment)`
+   - `DD_VERSION=(application version)`
```
Comment (alexandre-normand) [inline content/en/database_monitoring/guide/connect_dbm_and_apm.md:286]: Good callout. We probably should have this for all tracers as well.

Comment (zacharycmontoya) [reply content/en/database_monitoring/guide/connect_dbm_and_apm.md:286]: I stole this from the Python tab 😅

Comment (alexandre-normand) [reply content/en/database_monitoring/guide/connect_dbm_and_apm.md:286]: ah! 🙄 
It was a good thing to steal. We can follow-up later to maybe move this up but we don't have to do this now.

**Review by alexandre-normand — APPROVED** [review]

File: content/en/database_monitoring/guide/connect_dbm_and_apm.md [hunk 2]
```diff
@@ -44,8 +44,11 @@ Data privacy
 |          | [psycopg2][12]                 | {{< X >}}   |             |
 | **Java**     |                            |             |             |
 |          | jdbc                           | Coming soon | Coming soon |
-| **.NET**     |                            |             |             |
-|          |                                | Coming soon | Coming soon |
+| **.NET** [dd-trace-dotnet][15] >= 2.26.0 ||             |             |
```
Comment (EmmaOnThursday) [inline content/en/database_monitoring/guide/connect_dbm_and_apm.md:47]: could you move Java row to the end of the table to keep the `Coming soon` chunk last?

File: content/en/tracing/trace_collection/library_config/dotnet-core.md [hunk 3]
```diff
@@ -210,6 +210,10 @@ The following configuration variables are available **only** when using automati
 Enables or disables all automatic instrumentation. Setting the environment variable to `false` completely disables the CLR profiler. For other configuration methods, the CLR profiler is still loaded, but traces will not be generated. Valid values are: `true` or `false`.<br>
 **Default**: `true`
 
+`DD_DBM_PROPAGATION_MODE`
+: Enables DBM to APM link when set to `'service'` or `'full'`. The `'service'` option enables the connection between DBM and APM services. The `'full'` option enables connection between database spans with database query events. Available for Postgres and MySql.<br>
```
Comment (EmmaOnThursday) [inline content/en/tracing/trace_collection/library_config/dotnet-core.md:214]: Can we phrase this in a way that allows us to say `Database Monitoring` and link to the DBM docs in general? Something like `Enables linking between data sent from APM and the Database Monitoring product....` if that's not too wordy, with a link to https://docs.datadoghq.com/database_monitoring/

Comment (EmmaOnThursday) [reply content/en/tracing/trace_collection/library_config/dotnet-core.md:214]: (here & in the framework page as well)

Comment (alexandre-normand) [inline content/en/tracing/trace_collection/library_config/dotnet-core.md:214]: Oh, also the casing we use for mysql is `MySQL`.

Comment (zacharycmontoya) [reply content/en/tracing/trace_collection/library_config/dotnet-core.md:214]: One existing feature we have that links APM and Logs is a feature we tend to call "Log Correlation" (see the docs [here](https://docs.datadoghq.com/tracing/other_telemetry/connect_logs_and_traces/dotnet/?tab=serilog#configure-injection-in-logs)). Would some of that language be suitable here?

Comment (EmmaOnThursday) [reply content/en/tracing/trace_collection/library_config/dotnet-core.md:214]: oh wow, in general this is language we should steal for the main documentation page. I'll work on that separately. 

For this chunk, I don't think we need big changes to your language - just a way to insert the full phrase `Database Monitoring` with a link before we refer to it using the initials DBM. I think if we just changed the first sentence to `Enables linking between data sent from APM and the Database Monitoring product when set to 'service' or 'full'` that would do it.

Comment (zacharycmontoya) [reply content/en/tracing/trace_collection/library_config/dotnet-core.md:214]: > I think if we just changed the first sentence to Enables linking between data sent from APM and the Database Monitoring product when set to 'service' or 'full' that would do it.

Sounds good. I'll go ahead with this suggestion 👍🏼

Comment (zacharycmontoya) [reply content/en/tracing/trace_collection/library_config/dotnet-core.md]: Fixed in 69bfe6ece90c7bb51c1f11e37cc6e4d846f98d47

File: content/en/database_monitoring/guide/connect_dbm_and_apm.md (hunk 2 above)
Comment (zacharycmontoya) [reply content/en/database_monitoring/guide/connect_dbm_and_apm.md:45]: Fixed in 69bfe6ece90c7bb51c1f11e37cc6e4d846f98d47

File: content/en/tracing/trace_collection/library_config/dotnet-core.md (hunk 3 above)
Comment (zacharycmontoya) [reply content/en/tracing/trace_collection/library_config/dotnet-core.md]: Done in 69bfe6ece90c7bb51c1f11e37cc6e4d846f98d47

**Review by EmmaOnThursday — APPROVED** [review]

**Review by rtrieu — APPROVED** [review]
Minor change suggested, but otherwise looks good.
